whois-api/src/main/java/net/ripe/db/whois/api/httpserver/JettyBootstrap.java

```diff
@@ -258,33 +265,51 @@ private Connector createInsecureConnector(final Server server, final HttpConfigu
 
     /**
      * Use the DoSFilter from Jetty for rate limiting: https://www.eclipse.org/jetty/documentation/current/dos-filter.html.
-     * See {@link WhoisDoSFilter} for the customisations added.
-     * @return the rate limiting filter
+     * See {@link DoSLookUpFilter} or {@link DoSUpdateFilter} for the customisations added.
      * @throws JmxException if anything goes wrong JMX wise
      * @throws JMException if anything goes wrong JMX wise
      */
-    private FilterHolder createDosFilter() throws JmxException, JMException {
-        WhoisDoSFilter dosFilter = new WhoisDoSFilter();
-        FilterHolder holder = new FilterHolder(dosFilter);
-        holder.setName("DoSFilter");
-
+    private void createDosFilter(final WebAppContext context) throws JmxException, JMException {
         if (!dosFilterEnabled) {
             LOGGER.info("DoSFilter is *not* enabled");
+            return;
         }
+
+        final String maxRequestPerMsLookUp = "" + 10 * 60 * 1_000; // high default, 10 minutes
+        final FilterHolder lookUpDoSFilter = createDosFilter("DoSLookUpFilter", "50", maxRequestPerMsLookUp, new DoSLookUpFilter());
+
+        final FilterHolder updateDoSFilter = createDosFilter("DoSUpdateFilter", dosUpdatesMax, "" + Integer.parseInt(dosUpdatesMax) * 1_000, new DoSUpdateFilter());
+
+        if (!ManagementFactory.getPlatformMBeanServer().isRegistered(dosLookUpFilterMBeanName)) {
+            ManagementFactory.getPlatformMBeanServer().registerMBean(new ObjectMBean(lookUpDoSFilter), dosLookUpFilterMBeanName);
+        }
+
+        if (!ManagementFactory.getPlatformMBeanServer().isRegistered(dosUpdateFilterMBeanName)) {
+            ManagementFactory.getPlatformMBeanServer().registerMBean(new ObjectMBean(updateDoSFilter), dosUpdateFilterMBeanName);
+        }
+
+        context.addFilter(lookUpDoSFilter, "/*", EnumSet.allOf(DispatcherType.class));
+        context.addFilter(updateDoSFilter, "/*", EnumSet.allOf(DispatcherType.class));
+    }
+
+    private FilterHolder createDosFilter(final String dosFilterName, final String maxRequestsPerSecond, final String maxRequestsPerMs, final WhoisDoSFilter whoisDoSFilter) throws JmxException {
+        FilterHolder holder = new FilterHolder(whoisDoSFilter);
+        holder.setName(dosFilterName);
+
+        holder.setInitParameter("maxRequestMs", maxRequestsPerMs);
+        holder.setInitParameter("maxRequestsPerSec", maxRequestsPerSecond);
+        setCommonDoSFilterConfiguration(holder);
+
+        return holder;
+    }
+
+    private void setCommonDoSFilterConfiguration(final FilterHolder holder) {
         holder.setInitParameter("enabled", Boolean.toString(dosFilterEnabled));
-        holder.setInitParameter("maxRequestsPerSec", "50");
```

Comment:
These setters are way easier to read than using constructor arguments

Reply:
I refactor this, could you double check again?